### PR TITLE
feat: prague engine api types

### PR DESCRIPTION
**Pull request by onbjerg**
- Adds `engine_newPayloadV4`
- Adds `engine_getPayloadV4`
- Updates payload/payload attributes validation to handle V4 payloads

### CODE REVIEW

File: crates/rpc/rpc-types/src/beacon/payload.rs [hunk 1]
```diff
@@ -498,6 +498,10 @@ impl<'a> From<&'a ExecutionPayload> for BeaconExecutionPayload<'a> {
             ExecutionPayload::V3(payload) => {
                 BeaconExecutionPayload::V3(BeaconExecutionPayloadV3::from(payload))
             }
+            ExecutionPayload::V4(payload) => {
+                // TODO(onbjerg): Implement `ExecutionPayloadV4` support
+                todo!()
```
Comment (onbjerg) [inline crates/rpc/rpc-types/src/beacon/payload.rs:503]: for follow up since this is not strictly necessary for the engine api

**Comment by rkrasiuk** [comment]
@onbjerg still blocked?

**Comment by onbjerg** [comment]
nop

File: crates/rpc/rpc-types-compat/src/engine/payload.rs [hunk 2]
```diff
@@ -83,6 +84,15 @@ pub fn try_payload_v3_to_block(payload: ExecutionPayloadV3) -> Result<Block, Pay
...
+pub fn try_payload_v4_to_block(payload: ExecutionPayloadV4) -> Result<Block, PayloadError> {
+    // this performs the same conversion as the underlying V3 payload.
+    //
+    // the new request lists (`deposit_requests`, `withdrawal_requests`) are EL -> CL only, so we do
+    // not do anything special here to handle them
+    try_payload_v3_to_block(payload.payload_inner)
+}
+
 /// Converts [SealedBlock] to [ExecutionPayload]
 pub fn block_to_payload(value: SealedBlock) -> ExecutionPayload {
```
Comment (rkrasiuk) [inline crates/rpc/rpc-types-compat/src/engine/payload.rs:97]: this function needs adjustment as well

Comment (onbjerg) [reply crates/rpc/rpc-types-compat/src/engine/payload.rs:97]: fn seems unused?

**Review by rkrasiuk — APPROVED** [review]

File: crates/rpc/rpc-types-compat/src/engine/payload.rs (hunk 2 above)
Comment (onbjerg) [reply crates/rpc/rpc-types-compat/src/engine/payload.rs:97]: used in reorg sim, will adjust it in #8053

**Pull request merged**